code/modules/projectiles/ammunition/magazines.dm

```diff
@@ -114,6 +114,27 @@
 	ammo_type = /obj/item/ammo_casing/shotgun/improvised
 	max_ammo = 1
 
+/obj/item/ammo_box/magazine/internal/shot/improvised/New()
+ 	update_icon()
+
+/obj/item/ammo_box/magazine/internal/shot/improvised/i45
+	name = "pistol carbine internal magazine"
+	ammo_type = /obj/item/ammo_casing/c45
+	caliber = ".45"
+	max_ammo = 1
```

Comment:
I think 3-5 for the .45 wouldn't be too much either, but as a side though, we might want to add in the .45 rubber ammo to autolathes.